### PR TITLE
Update assign operator

### PR DESCRIPTION
**Pull request by achoum**
Please review from PR by comparing it to branch `gbm_plotting`

### CODE REVIEW

File: temporian/core/operators/assign.py [hunk 1]
```diff
@@ -22,34 +24,50 @@
...
+    """Assign operator."""
 
     def __init__(
         self,
-        left_event: Event,
-        right_event: Event,
+        event_1: Event,
+        event_2: Event,
+        event_3: Optional[Event] = None,
+        event_4: Optional[Event] = None,
```
Comment (diegomarvid) [inline temporian/core/operators/assign.py:34]: Why do we have 2 optional events to assign and not a list of optional events?

**Review by diegomarvid — APPROVED** [review]
LGTM! Great work Mathieu 🎉

File: temporian/core/operators/assign.py (hunk 1 above)
Comment (ianspektor) [reply temporian/core/operators/assign.py:34]: Agree! using `*args` here to accept an arbitrary number of events would both improve the API and simplify the code below

Comment (achoum) [reply temporian/core/operators/assign.py:34]: Good point.
*Done

File: temporian/test/prototype_test_numpy.py [hunk 2]
```diff
@@ -76,75 +76,56 @@ def setUp(self) -> None:
...
+                self.event_2: self.event_2_data,
             },
             # TODO: The assign operator has some issues with dtypes. Re-enable
             # checking when solved.
-            check_execution=False,
+            check_execution=True,
         )
 
-        # validate
+        print(output_event_numpy)
```
[the remark this answers is not in the document]
Comment (achoum) [reply temporian/test/prototype_test_numpy.py:127]: *Done

File: temporian/core/operators/assign.py [hunk 3]
```diff
@@ -59,8 +77,10 @@ def build_op_definition(cls) -> pb.OperatorDef:
         return pb.OperatorDef(
             key="ASSIGN",
             inputs=[
-                pb.OperatorDef.Input(key="left_event"),
-                pb.OperatorDef.Input(key="right_event"),
+                pb.OperatorDef.Input(key="event_1"),
+                pb.OperatorDef.Input(key="event_2"),
+                pb.OperatorDef.Input(key="event_3", is_optional=True),
+                pb.OperatorDef.Input(key="event_4", is_optional=True),
```
[the remark this answers is not in the document]
Comment (achoum) [reply temporian/core/operators/assign.py:83]: Yes. This is the reason. Arguments checking and serialization also depends on it.
I updated the code to no list the inputs.

**Pull request merged**